pandas/core/arrays/period.py

```diff
@@ -73,6 +73,9 @@ def wrapper(self, other):
                 msg = DIFFERENT_FREQ_INDEX.format(self.freqstr, other.freqstr)
                 raise IncompatibleFrequency(msg)
 
+            if other.ndim > 0 and len(self) != len(other):
```

Comment:
this should fail if ndim of other != 1

we don't generally check this for other arrays (though we should). So I would maybe add this check on the base array class and just call it here, ``_validate_equal(self, other)`` or something

Reply:
yes, I think we have an implicit guarantee on 1-D ness now (or at the very least the pandas extension types do).

Reply:
Related #23853